webrender/src/tiling.rs

```diff
@@ -2608,103 +2613,143 @@ impl FrameBuilder {
+                    if layer.is_visible() {
                         let auxiliary_lists = pipeline_auxiliary_lists.get(&layer.pipeline_id)
                                                                       .expect("No auxiliary lists?!");
                         let layer_world_clip_rect = layer.world_clip_rect.unwrap();
 
-                        for item in &mut layer.items {
-                            match item {
-                                &mut StackingContextItem::StackingContext(..) => {
-                                    // TODO(gw): Worth removing these to reduce cmd list size?
-                                }
-                                &mut StackingContextItem::Primitive(prim_index) => {
-                                    let prim = &mut self.prim_store[prim_index.0];
-
-                                    prim.bounding_rect = None;
-
-                                    let local_rect = prim.rect.intersection(&prim.local_clip_rect);
-
-                                    if let Some(local_rect) = local_rect {
-                                        let xf_rect = TransformedRect::new(&local_rect,
-                                                                           &layer.transform,
-                                                                           self.device_pixel_ratio);
-
-                                        let prim_rect = xf_rect.bounding_rect
-                                                               .intersection(&layer_world_clip_rect);
-                                        if let Some(prim_rect) = prim_rect {
-                                            if prim_rect.intersects(&screen_rect) {
-                                                prim.bounding_rect = Some(prim_rect);
-                                                if prim.build_resource_list(resource_list, auxiliary_lists) {
-                                                    layer.prims_to_prepare.push(prim_index);
-                                                }
-                                            }
+                        for i in 0..prim_count {
+                            let prim_index = PrimitiveIndex(prim_index.0 + i);
+
+                            let prim = &mut self.prim_store[prim_index.0];
+                            prim.bounding_rect = None;
+
+                            let local_rect = prim.rect.intersection(&prim.local_clip_rect);
+
+                            if let Some(local_rect) = local_rect {
+                                let xf_rect = TransformedRect::new(&local_rect,
+                                                                   &layer.transform,
+                                                                   self.device_pixel_ratio);
+
+                                let prim_rect = xf_rect.bounding_rect.intersection(&layer_world_clip_rect);
+
+                                if let Some(prim_rect) = prim_rect {
```

Comment:
Up to you, but could be:

    let prim_rect = match prim_rect {
        Some(prim_rect) if prim_rect.intersects(&screen_rect) => prim_rect,
        Some(_) | None => continue,
    };